src/vs/workbench/api/browser/mainThreadFileSystemEventService.ts

```diff
@@ -74,8 +74,8 @@ export class MainThreadFileSystemEventService {
 
 
 		const fileOperationParticipant = new class implements IWorkingCopyFileOperationParticipant {
-			async participate(files: SourceTargetPair[], operation: FileOperation, undoRedoGroupId: number | undefined, isUndoing: boolean | undefined, timeout: number, token: CancellationToken) {
-				if (isUndoing) {
+			async participate(files: SourceTargetPairWithOptions[], operation: FileOperation, timeout: number, token: CancellationToken) {
+				if (files.some(f => f.options?.isUndoing)) {
```

Comment:
My feeling is that some options, like `isUndoing`, aren't part of the `SourceTargetPairWithOptions`. Like the operation is undoing 3 creations etc, so isUnding and the undoRepo groups belong is a single options bag if you ask me.

Reply:
This suggestoin makes sense to me. I will go down this path.
@bpasero fyi